### PR TITLE
Remove lambdas to reduce number of runtime classes.

### PR DESCRIPTION
**Pull request by radcortez**
For https://github.com/quarkusio/quarkus/issues/12960.

We can probably be even more aggressive and remove a few more, but I think we will require a couple of API tweaks that I don't want to introduce now. 

This is how it looks now:

```
io.smallrye.config.ConfigMapping$$ProxyImpl
io.smallrye.config.ConfigMappingContext
io.smallrye.config.ConfigMappingContext$$Lambda$05b87c14bf80d38480b3cf6d7a93e1666f21625d
io.smallrye.config.ConfigMappingContext$$Lambda$3e4aec83af6e494147701818f695759ec154b2d2
io.smallrye.config.ConfigMappingContext$$Lambda$597eb83f410793934bed538f6394eea992dbb70d
io.smallrye.config.ConfigMappingContext$$Lambda$6ed21dc2de9600cb66a9a01a6718eef660dee9ed
io.smallrye.config.ConfigMappingContext$$Lambda$7cd4c6819fec0cb86628dcb54becbf262c41d285
io.smallrye.config.ConfigMappingContext$$Lambda$7cd4c6819fec0cb86628dcb54becbf262c41d285_1
io.smallrye.config.ConfigMappingContext$$Lambda$8a9157a1f7543b34cf034a5381a69dec3d10c31b
io.smallrye.config.ConfigMappingContext$$Lambda$929cc83ba8d650a0dacf1a38806459b85ea3d8a2
io.smallrye.config.ConfigMappingContext$$Lambda$b029b2def269c72ddd3c2c4f69a97752f2d3ed2c
io.smallrye.config.ConfigMappingContext$$Lambda$c3d2d3ee5a351f40066a3e735511cf184268cf88
io.smallrye.config.ConfigMappingContext$$Lambda$c3d2d3ee5a351f40066a3e735511cf184268cf88_1
io.smallrye.config.ConfigMappingContext$$Lambda$e2137bf87fd5e433fc786fb456da637fe6bc4c3d
io.smallrye.config.ConfigMappingInterface
io.smallrye.config.ConfigMappingInterface$1
io.smallrye.config.ConfigMappingInterface$GroupProperty
io.smallrye.config.ConfigMappingInterface$LeafProperty
io.smallrye.config.ConfigMappingInterface$MapProperty
io.smallrye.config.ConfigMappingInterface$MayBeOptionalProperty
io.smallrye.config.ConfigMappingInterface$OptionalProperty
io.smallrye.config.ConfigMappingInterface$PrimitiveProperty
io.smallrye.config.ConfigMappingInterface$Property
io.smallrye.config.ConfigMappingLoader
io.smallrye.config.ConfigMappingLoader$1
io.smallrye.config.ConfigMappingLoader$ConfigMappingObjectHolder
io.smallrye.config.ConfigMappingProvider
io.smallrye.config.ConfigMappingProvider$$Lambda$1c471ae38e73c0207060ca58c6ad6a19e48a4a1e
io.smallrye.config.ConfigMappingProvider$$Lambda$a3219019b0b0ed4e4b026ca3dd004ceb5db622eb
io.smallrye.config.ConfigMappingProvider$$Lambda$af395d2fcb657cdb6b7d7a90b938c3ac4603079b
io.smallrye.config.ConfigMappingProvider$Builder
io.smallrye.config.ConfigMappingProvider$Builder$$Lambda$d67c0450a6ff3ac39b78135d421d7e0edf73cdff
io.smallrye.config.ConfigMappingProvider$ConsumeOneAndThen
io.smallrye.config.ConfigMappingProvider$ConsumeOneAndThenFn
io.smallrye.config.ConfigMappingProvider$GetFieldOfEnclosing
io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInGroup
io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInMap
io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingMapInGroup
io.smallrye.config.ConfigMappingProvider$GetRootAction
io.smallrye.config.ConfigMappings
io.smallrye.config.ConfigMappings$ConfigMappingWithPrefix
io.smallrye.config.ConfigMessages_$bundle
io.smallrye.config.ConfigSourceInterceptor
io.smallrye.config.ConfigSourceInterceptor$1
io.smallrye.config.ConfigSourceInterceptorFactory
io.smallrye.config.ConfigValidationException
io.smallrye.config.ConfigValidationException$Problem
io.smallrye.config.ConfigValue
io.smallrye.config.ConfigValue$ConfigValueBuilder
io.smallrye.config.ConfigValueConfigSourceWrapper
io.smallrye.config.ConfigValueConverter
io.smallrye.config.ConfigurableConfigSource
io.smallrye.config.Converters
io.smallrye.config.Converters$$Lambda$192f713b88f8c6a0b949ed761eabc7c44eb22b15
io.smallrye.config.Converters$$Lambda$3102b341117f732b9af1cbbf44af7099bbe7fc2d
io.smallrye.config.Converters$$Lambda$45a62084808f9a509e234582cd3f303137997fce
io.smallrye.config.Converters$$Lambda$47a15b4c889e62e4078c9d1cc5b41f96e8813158
io.smallrye.config.Converters$$Lambda$62633287c3fb8ab4e45ec37b65e3233cc28240ce
io.smallrye.config.Converters$$Lambda$62e4f113a6b2364a0408a99e8420024085009f03
io.smallrye.config.Converters$$Lambda$93822c9b901deba3c19618f88cd7af53d9bae9ee
io.smallrye.config.Converters$$Lambda$9fd315a515a761e75715822babd55e45f6be436c
io.smallrye.config.Converters$$Lambda$ac4cd72addf71a4e652c2330dda9e68db0f7df44
io.smallrye.config.Converters$$Lambda$c269514b070900e3269af8e3300f7abb68e1892d
io.smallrye.config.Converters$$Lambda$c367f1a32eba16a0d325faa5b60583ae976e94e9
io.smallrye.config.Converters$$Lambda$d4aad569ff953d371ae5cf05fd9f24cea6014c1f
io.smallrye.config.Converters$ArrayConverter
io.smallrye.config.Converters$BuiltInConverter
io.smallrye.config.Converters$CollectionConverter
io.smallrye.config.Converters$EmptyValueConverter
io.smallrye.config.Converters$OptionalConverter
io.smallrye.config.Converters$OptionalDoubleConverter
io.smallrye.config.Converters$OptionalIntConverter
io.smallrye.config.Converters$OptionalLongConverter
io.smallrye.config.Converters$TrimmingConverter
io.smallrye.config.DefaultValuesConfigSource
io.smallrye.config.EnvConfigSource
io.smallrye.config.EnvConfigSource$$Lambda$3d9b810a5b232c8f4224970f4b200f104ede745f
io.smallrye.config.ExpressionConfigSourceInterceptor
io.smallrye.config.ExpressionConfigSourceInterceptor$$Lambda$dd4309abcc9dfe5c68e9e1c89c058970515edf06
io.smallrye.config.Expressions
io.smallrye.config.ImplicitConverters
io.smallrye.config.ImplicitConverters$ConstructorConverter
io.smallrye.config.ImplicitConverters$StaticMethodConverter
io.smallrye.config.KeyMap
io.smallrye.config.KeyMap$$Lambda$29a8250fbdd7c7166ebb61c07d6297771e747440
io.smallrye.config.KeyMap$$Lambda$ef64897b27106ac8c471c389d8d2818731466e5c
io.smallrye.config.KeyMapBackedConfigSource
io.smallrye.config.NameIterator
io.smallrye.config.ProfileConfigSourceInterceptor
io.smallrye.config.ProfileConfigSourceInterceptor$$Lambda$7cd4c6819fec0cb86628dcb54becbf262c41d285
io.smallrye.config.ProfileConfigSourceInterceptor$$Lambda$c8eb451f8d060946083f3fb1bc2379f44aaf0f82
io.smallrye.config.PropertiesConfigSource
io.smallrye.config.PropertiesConfigSourceProvider
io.smallrye.config.SecretKeys
io.smallrye.config.SecretKeysConfigSourceInterceptor
io.smallrye.config.SecuritySupport
io.smallrye.config.SmallRyeConfig
io.smallrye.config.SmallRyeConfig$$Lambda$075d695c3b2815d3d1054cc789ded21707e71719
io.smallrye.config.SmallRyeConfig$$Lambda$a0d92915a8367f555357708a5359de2231db64a4
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority$$Lambda$3d47c1832f900f096e85ab3da6c2d06c11158ffc
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority$$Lambda$93a94517884c0f8e48f48eba278e7256cfc93ead
io.smallrye.config.SmallRyeConfig$ConfigSources
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$206f62e681a79840a48ceb893bf6a4d4852fbc92
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$206f62e681a79840a48ceb893bf6a4d4852fbc92_1
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$e9a847c7ff5d8ebb1487513b656dc84940c0398f
io.smallrye.config.SmallRyeConfig$ConfigSources$1
io.smallrye.config.SmallRyeConfigBuilder
io.smallrye.config.SmallRyeConfigBuilder$1
io.smallrye.config.SmallRyeConfigBuilder$ConverterWithPriority
io.smallrye.config.SmallRyeConfigBuilder$InterceptorWithPriority
io.smallrye.config.SmallRyeConfigBuilder$InterceptorWithPriority$1
io.smallrye.config.SmallRyeConfigProviderResolver
io.smallrye.config.SmallRyeConfigSourceInterceptor
io.smallrye.config.SmallRyeConfigSourceInterceptorContext
io.smallrye.config.SysPropConfigSource
io.smallrye.config.SysPropConfigSource$$Lambda$14251524d5c32868980515f48e14b1696bd81b34
io.smallrye.config.SysPropConfigSource$$Lambda$db9e7089470a716c48d2693f3b42683e48f3bccc
io.smallrye.config.WithConverter$$ProxyImpl
io.smallrye.config.WithDefault$$ProxyImpl
io.smallrye.config.WithName$$ProxyImpl
io.smallrye.config.common.AbstractConfigSource
io.smallrye.config.common.AbstractConverter
io.smallrye.config.common.AbstractDelegatingConverter
io.smallrye.config.common.MapBackedConfigSource
io.smallrye.config.common.utils.ConfigSourceUtil
io.smallrye.config.common.utils.StringUtil
io.smallrye.config.inject.ConfigProducer
io.smallrye.config.inject.ConfigProducer_Bean
io.smallrye.config.inject.ConfigProducer_ClientProxy
io.smallrye.config.inject.ConfigProducer_ProducerMethod_getConfig_3239e733560478b1cd66965e2a7b33065ff9aa4c_Bean
io.smallrye.config.inject.InjectionMessages_$bundle
io.smallrye.config.inject.SecuritySupport
```

And before:

```
io.smallrye.config.ConfigMapping$$ProxyImpl
io.smallrye.config.ConfigMappingContext
io.smallrye.config.ConfigMappingContext$$Lambda$05b87c14bf80d38480b3cf6d7a93e1666f21625d
io.smallrye.config.ConfigMappingContext$$Lambda$3e4aec83af6e494147701818f695759ec154b2d2
io.smallrye.config.ConfigMappingContext$$Lambda$597eb83f410793934bed538f6394eea992dbb70d
io.smallrye.config.ConfigMappingContext$$Lambda$6ed21dc2de9600cb66a9a01a6718eef660dee9ed
io.smallrye.config.ConfigMappingContext$$Lambda$7cd4c6819fec0cb86628dcb54becbf262c41d285
io.smallrye.config.ConfigMappingContext$$Lambda$7cd4c6819fec0cb86628dcb54becbf262c41d285_1
io.smallrye.config.ConfigMappingContext$$Lambda$8a9157a1f7543b34cf034a5381a69dec3d10c31b
io.smallrye.config.ConfigMappingContext$$Lambda$929cc83ba8d650a0dacf1a38806459b85ea3d8a2
io.smallrye.config.ConfigMappingContext$$Lambda$b029b2def269c72ddd3c2c4f69a97752f2d3ed2c
io.smallrye.config.ConfigMappingContext$$Lambda$c3d2d3ee5a351f40066a3e735511cf184268cf88
io.smallrye.config.ConfigMappingContext$$Lambda$c3d2d3ee5a351f40066a3e735511cf184268cf88_1
io.smallrye.config.ConfigMappingContext$$Lambda$e2137bf87fd5e433fc786fb456da637fe6bc4c3d
io.smallrye.config.ConfigMappingInterface
io.smallrye.config.ConfigMappingInterface$$Lambda$0a4a0881e414c5eb52f63f00bdcd5bdbb1ceaa11
io.smallrye.config.ConfigMappingInterface$$Lambda$4807525a9f61549928b328a39347420f40b7ca10
io.smallrye.config.ConfigMappingInterface$1
io.smallrye.config.ConfigMappingInterface$GroupProperty
io.smallrye.config.ConfigMappingInterface$LeafProperty
io.smallrye.config.ConfigMappingInterface$MapProperty
io.smallrye.config.ConfigMappingInterface$MayBeOptionalProperty
io.smallrye.config.ConfigMappingInterface$OptionalProperty
io.smallrye.config.ConfigMappingInterface$PrimitiveProperty
io.smallrye.config.ConfigMappingInterface$Property
io.smallrye.config.ConfigMappingLoader
io.smallrye.config.ConfigMappingLoader$1
io.smallrye.config.ConfigMappingLoader$ConfigMappingObjectHolder
io.smallrye.config.ConfigMappingProvider
io.smallrye.config.ConfigMappingProvider$$Lambda$1c471ae38e73c0207060ca58c6ad6a19e48a4a1e
io.smallrye.config.ConfigMappingProvider$$Lambda$a3219019b0b0ed4e4b026ca3dd004ceb5db622eb
io.smallrye.config.ConfigMappingProvider$$Lambda$af395d2fcb657cdb6b7d7a90b938c3ac4603079b
io.smallrye.config.ConfigMappingProvider$Builder
io.smallrye.config.ConfigMappingProvider$Builder$$Lambda$d67c0450a6ff3ac39b78135d421d7e0edf73cdff
io.smallrye.config.ConfigMappingProvider$ConsumeOneAndThen
io.smallrye.config.ConfigMappingProvider$ConsumeOneAndThenFn
io.smallrye.config.ConfigMappingProvider$GetFieldOfEnclosing
io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInGroup
io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingGroupInMap
io.smallrye.config.ConfigMappingProvider$GetOrCreateEnclosingMapInGroup
io.smallrye.config.ConfigMappingProvider$GetRootAction
io.smallrye.config.ConfigMappings
io.smallrye.config.ConfigMappings$$Lambda$705667d752d3acfb06cd8e498d27d9ae6b9a544e
io.smallrye.config.ConfigMappings$ConfigMappingWithPrefix
io.smallrye.config.ConfigMessages_$bundle
io.smallrye.config.ConfigSourceInterceptor
io.smallrye.config.ConfigSourceInterceptor$1
io.smallrye.config.ConfigSourceInterceptorFactory
io.smallrye.config.ConfigValidationException
io.smallrye.config.ConfigValidationException$Problem
io.smallrye.config.ConfigValue
io.smallrye.config.ConfigValue$ConfigValueBuilder
io.smallrye.config.ConfigValueConfigSourceWrapper
io.smallrye.config.ConfigValueConverter
io.smallrye.config.ConfigurableConfigSource
io.smallrye.config.Converters
io.smallrye.config.Converters$$Lambda$192f713b88f8c6a0b949ed761eabc7c44eb22b15
io.smallrye.config.Converters$$Lambda$3102b341117f732b9af1cbbf44af7099bbe7fc2d
io.smallrye.config.Converters$$Lambda$45a62084808f9a509e234582cd3f303137997fce
io.smallrye.config.Converters$$Lambda$47a15b4c889e62e4078c9d1cc5b41f96e8813158
io.smallrye.config.Converters$$Lambda$62633287c3fb8ab4e45ec37b65e3233cc28240ce
io.smallrye.config.Converters$$Lambda$62e4f113a6b2364a0408a99e8420024085009f03
io.smallrye.config.Converters$$Lambda$93822c9b901deba3c19618f88cd7af53d9bae9ee
io.smallrye.config.Converters$$Lambda$9fd315a515a761e75715822babd55e45f6be436c
io.smallrye.config.Converters$$Lambda$ac4cd72addf71a4e652c2330dda9e68db0f7df44
io.smallrye.config.Converters$$Lambda$c269514b070900e3269af8e3300f7abb68e1892d
io.smallrye.config.Converters$$Lambda$c367f1a32eba16a0d325faa5b60583ae976e94e9
io.smallrye.config.Converters$$Lambda$d4aad569ff953d371ae5cf05fd9f24cea6014c1f
io.smallrye.config.Converters$ArrayConverter
io.smallrye.config.Converters$BuiltInConverter
io.smallrye.config.Converters$CollectionConverter
io.smallrye.config.Converters$EmptyValueConverter
io.smallrye.config.Converters$OptionalConverter
io.smallrye.config.Converters$OptionalDoubleConverter
io.smallrye.config.Converters$OptionalIntConverter
io.smallrye.config.Converters$OptionalLongConverter
io.smallrye.config.Converters$TrimmingConverter
io.smallrye.config.DefaultValuesConfigSource
io.smallrye.config.DefaultValuesConfigSource$$Lambda$80787c7ca1de64c8a1523b65783ddc0c27248f35
io.smallrye.config.EnvConfigSource
io.smallrye.config.EnvConfigSource$$Lambda$3d9b810a5b232c8f4224970f4b200f104ede745f
io.smallrye.config.ExpressionConfigSourceInterceptor
io.smallrye.config.ExpressionConfigSourceInterceptor$$Lambda$dd4309abcc9dfe5c68e9e1c89c058970515edf06
io.smallrye.config.Expressions
io.smallrye.config.ImplicitConverters
io.smallrye.config.ImplicitConverters$ConstructorConverter
io.smallrye.config.ImplicitConverters$StaticMethodConverter
io.smallrye.config.KeyMap
io.smallrye.config.KeyMap$$Lambda$29a8250fbdd7c7166ebb61c07d6297771e747440
io.smallrye.config.KeyMap$$Lambda$ef64897b27106ac8c471c389d8d2818731466e5c
io.smallrye.config.KeyMapBackedConfigSource
io.smallrye.config.NameIterator
io.smallrye.config.ProfileConfigSourceInterceptor
io.smallrye.config.ProfileConfigSourceInterceptor$$Lambda$140aebe0ff276cabdfa3e45528121c9e69e6b00d
io.smallrye.config.ProfileConfigSourceInterceptor$$Lambda$7cd4c6819fec0cb86628dcb54becbf262c41d285
io.smallrye.config.ProfileConfigSourceInterceptor$$Lambda$c8eb451f8d060946083f3fb1bc2379f44aaf0f82
io.smallrye.config.ProfileConfigSourceInterceptor$$Lambda$f390b3ef9afee65763558f1264bfe5adf8d8f353
io.smallrye.config.PropertiesConfigSource
io.smallrye.config.PropertiesConfigSourceProvider
io.smallrye.config.SecretKeys
io.smallrye.config.SecretKeysConfigSourceInterceptor
io.smallrye.config.SecuritySupport
io.smallrye.config.SmallRyeConfig
io.smallrye.config.SmallRyeConfig$$Lambda$03289db87770c2ce6990c2bdfc1ee6477c0fabbf
io.smallrye.config.SmallRyeConfig$$Lambda$070589a755eb0b18e7db8f5f1365ce2028dc3d02
io.smallrye.config.SmallRyeConfig$$Lambda$12154388409ea0d7dbb38539c1040e8fa4e1c79e
io.smallrye.config.SmallRyeConfig$$Lambda$a5e20c33da2c7ea09ccf803d8524fb9b0eab674e
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority$$Lambda$30e350b4e2308284c5012645b1e849a9d20df527
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority$$Lambda$3d47c1832f900f096e85ab3da6c2d06c11158ffc
io.smallrye.config.SmallRyeConfig$ConfigSourceInterceptorWithPriority$$Lambda$93a94517884c0f8e48f48eba278e7256cfc93ead
io.smallrye.config.SmallRyeConfig$ConfigSources
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$182906ded383582afdf1d3b8d441290e08f30a96
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$206f62e681a79840a48ceb893bf6a4d4852fbc92
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$206f62e681a79840a48ceb893bf6a4d4852fbc92_1
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$28a06398a8f2809d1cba38d205e358aceebe1a2e
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$28a06398a8f2809d1cba38d205e358aceebe1a2e_1
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$2b23a6c3bcd0740890df1d749ed6c18a22a8dc22
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$482dce0acb4910dfcc36b1a8ba55dcb5f0532044
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$83b4cdd02e35157a531b0b823f862b6a6cb62f1d
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$cd172a24059a7a3956d8426ff35ffb3cc3756c49
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$cd172a24059a7a3956d8426ff35ffb3cc3756c49_1
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$e88526289073b4c0866f1962389b0b5afe7eebf7
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$e88526289073b4c0866f1962389b0b5afe7eebf7_1
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$e9a847c7ff5d8ebb1487513b656dc84940c0398f
io.smallrye.config.SmallRyeConfig$ConfigSources$$Lambda$fbf8cd4da9e271a576a2d6f048d819aa7c664cd3
io.smallrye.config.SmallRyeConfig$ConfigSources$1
io.smallrye.config.SmallRyeConfigBuilder
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$2be95fe9c729a9e458197894ef7c7cfe45b84656
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$539ae63109f7aaaf6f6e72f91595d205ca8974c2
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$6be3f47d066e1c46f3b81a65307185db9531be63
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$86e1704e667d452c667e52a4abd9fb44e282dbad
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$ba797fca8bd399214c854bec13715409e73fbab3
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$ba797fca8bd399214c854bec13715409e73fbab3_1
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$ba797fca8bd399214c854bec13715409e73fbab3_2
io.smallrye.config.SmallRyeConfigBuilder$$Lambda$e2660e70657a49c23a075290d2a34153cbe248dd
io.smallrye.config.SmallRyeConfigBuilder$1
io.smallrye.config.SmallRyeConfigBuilder$ConverterWithPriority
io.smallrye.config.SmallRyeConfigBuilder$InterceptorWithPriority
io.smallrye.config.SmallRyeConfigBuilder$InterceptorWithPriority$1
io.smallrye.config.SmallRyeConfigBuilder$InterceptorWithPriority$1$$Lambda$1534624681981c3272cd937f975598ed08ff1ef5
io.smallrye.config.SmallRyeConfigProviderResolver
io.smallrye.config.SmallRyeConfigSourceInterceptor
io.smallrye.config.SmallRyeConfigSourceInterceptor$$Lambda$c76b788b31f27e2aff91eef27a77a70f36808f2f
io.smallrye.config.SmallRyeConfigSourceInterceptorContext
io.smallrye.config.SysPropConfigSource
io.smallrye.config.SysPropConfigSource$$Lambda$14251524d5c32868980515f48e14b1696bd81b34
io.smallrye.config.SysPropConfigSource$$Lambda$db9e7089470a716c48d2693f3b42683e48f3bccc
io.smallrye.config.WithConverter$$ProxyImpl
io.smallrye.config.WithDefault$$ProxyImpl
io.smallrye.config.WithName$$ProxyImpl
io.smallrye.config.common.AbstractConfigSource
io.smallrye.config.common.AbstractConverter
io.smallrye.config.common.AbstractDelegatingConverter
io.smallrye.config.common.MapBackedConfigSource
io.smallrye.config.common.utils.ConfigSourceUtil
io.smallrye.config.common.utils.ConfigSourceUtil$$Lambda$0be0653192d680ff362f84852709f138568c306c
io.smallrye.config.common.utils.ConfigSourceUtil$$Lambda$51fa49c3f5ede21485f50c32bc2dfc7dbaab2716
io.smallrye.config.common.utils.StringUtil
io.smallrye.config.inject.ConfigProducer
io.smallrye.config.inject.ConfigProducer_Bean
io.smallrye.config.inject.ConfigProducer_ClientProxy
io.smallrye.config.inject.ConfigProducer_ProducerMethod_getConfig_3239e733560478b1cd66965e2a7b33065ff9aa4c_Bean
io.smallrye.config.inject.InjectionMessages_$bundle
io.smallrye.config.inject.SecuritySupport
```

### CODE REVIEW

**Review by gastaldi — DISMISSED** [review]
Nice! Didn't know lambdas produced so much code

**Comment by radcortez** [comment]
> Nice! Didn't know lambdas produced so much code

Just added a before list for a better comparison.

**Comment by gastaldi** [comment]
Can we remove lambdas completely from the list above?

File: implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java [hunk 1]
```diff
@@ -184,23 +196,27 @@ public SmallRyeConfigBuilder withSources(Collection<ConfigSource> configSources)
...
 
     public SmallRyeConfigBuilder withInterceptors(ConfigSourceInterceptor... interceptors) {
-        this.interceptors.addAll(Stream.of(interceptors)
-                .map(InterceptorWithPriority::new)
-                .collect(Collectors.toList()));
+        List<InterceptorWithPriority> interceptorWithPriorities = new ArrayList<>();
+        for (ConfigSourceInterceptor interceptor : interceptors) {
+            interceptorWithPriorities.add(new InterceptorWithPriority(interceptor));
+        }
+        this.interceptors.addAll(interceptorWithPriorities);
```
Comment (gsmet) [inline implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java:210]: Is there a reason why you introduce an intermediary list here?

File: implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java [hunk 2]
```diff
@@ -184,23 +196,27 @@ public SmallRyeConfigBuilder withSources(Collection<ConfigSource> configSources)
...
         return this;
     }
 
     public SmallRyeConfigBuilder withInterceptorFactories(ConfigSourceInterceptorFactory... interceptorFactories) {
-        this.interceptors.addAll(Stream.of(interceptorFactories)
-                .map(InterceptorWithPriority::new)
-                .collect(Collectors.toList()));
+        List<InterceptorWithPriority> interceptorWithPriorities = new ArrayList<>();
```
Comment (gsmet) [inline implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java:215]: Same question here.

**Review by gsmet — COMMENTED** [review]
Added some questions.

File: implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java (hunk 1 above)
Comment (radcortez) [reply implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java:210]: Yes, no point.

File: implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java (hunk 2 above)
Comment (radcortez) [reply implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java:215]: Yes, no point.

**Comment by radcortez** [comment]
> Can we remove lambdas completely from the list above?

Probably. Some would just be substituted by anonymous inner classes. Others we could completely remove, but require some more adjustments to their APIs. I didn't want to go too deep and risk introducing a breaking change.

Additional work will be put into these and I'll be more cautious from now on.

**Review by dmlloyd — APPROVED** [review]

**Review by gsmet — APPROVED** [review]
Looks good now, thanks!

**Comment by gsmet** [comment]
BTW, I'm not totally against lambdas when they make the code significantly better. But I would like us to avoid them when we can and it doesn't bring much.

**Comment by radcortez** [comment]
> BTW, I'm not totally against lambdas when they make the code significantly better. But I would like us to avoid them when we can and it doesn't bring much.

Yes, I confess there was some abuse here :) I'll try to keep this under control now.

**Comment by gsmet** [comment]
Perfect, thanks!

**Pull request merged**